### DISCUSSION TITLE
[General]: REQUEST PERMISSION

### DISCUSSION THREAD

**Discussion by frankmckale** [davidjerleke/embla-carousel#940] [Help] [answered]
### Summary

I would like to be able to use some functions from EmblaCarousel, specifically like.. DragHandler, DragTracker and Animations only. I intend to inject them into my own library. Ive read that Embla Carousel is open-source and allows modifications, but I'm seeking clarification on whether I can use parts of this library in my own library. I would appreciate it if I get permission. Thanks.

### If applicable, which variants of Embla Carousel are relevant to this conversation?

- [X] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment]
Hi @frankmckale,

I’m on vacation and will answer you when possible. Thanks for your patience.

**Reply by frankmckale** [reply]
its okay, enjoy your vacation 👍 :)

**Reply by davidjerleke** [reply, marked as answer]
> I'm seeking clarification on whether I can use parts of this library in my own library.

@frankmckale this project is [MIT licensed](https://opensource.org/license/mit), so you can pretty much do anything with it. However, the only **two things** you need to know are the following:

1. Copyright applies, so you can't use parts of this library in your own library and claim that these parts were made by you.
2. The below **copyright notice** and the **permission notice** shall be included in all copies or substantial portions of the Software:

### Copyright notice
Copyright © 2019-present, David Jerleke

### Permission notice
Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the “Software”), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so.

### My own words
At the time of writing, the sponsors/donations for this project cover a single 40 hour work week. And I've spent far more hours/weeks/months/years on this project so the bottom line is: I've practically made everything for free for other devs to use without any expectations of getting compensation for my work. So It's not mandatory in any way, but if someone wants to use the whole library or parts of it in their own library, I think it's good practice to say thanks by including a thanks notice in their README and/or website for their own library. 

Let me know if this clarifies things and if you have any follow up questions. Good luck with your own library!

Best,
David

**Reply by frankmckale** [reply]
Thanks, I'm a student, and I've been able to learn a lot more about JavaScript thanks to your library. I developed several carousel scripts using CSS animations and requestAnimationFrame. Gestures weren't a problem, but I did face issues such as a sensation of lag in requestAnimationFrame-based animations, and with CSS-based animations I struggled when I capture and drag it while its animating

With my limited knowledge, I took the "Animations, DragHandler, and DragTracker" functions from Embla Carousel to analyze and adapt them for my own library, maintaining the logic of how they were programmed. As you mentioned about copyright, I will properly apply copyright notices and permissions.

I would like to contribute to this library, but my skills are basic. It took me time to break down Embla Carousel using vanilla JavaScript to get DragHandler, DragTracker, and Animations. I'm developing a simple horizontal library focused on transitions between sections, similar to apps like FacebookAPK, WhatsAppAPK, or TikTokAPK, where you navigate between sections by hiding completly intermediate elements — essentially a "pager" with section jumps.

The Embla library was the most understandable for me and closer to a native Android touch carousels/pagers. Embla Carousel helped me better understand the logic of how to achieve this and make these enhancements, so I appreciate that your library has helped me a lot how to program a carousel. It's important to mention that my use of the library is personal and intended to expand my knowledge and present my thesis for study. 

Though if something arise from this work, you will be duly recognized and compensated. I have no intention of taking advantage or profiting from others' efforts, thats why I asked permissions first. 

Thank you so much David.

**Reply by davidjerleke** [reply]
@frankmckale thank you for sharing your use case. And I'm happy to hear that you find the Embla code most understandable, thanks for sharing that 👍. 

> Though if something arise from this work, you will be duly recognized and compensated. I have no intention of taking advantage or profiting from others' efforts, thats why I asked permissions first.

No worries! I just wanted to share the **My own words** section just in case you were to build something public. And another reason I included it in this discussion is because the information might be useful to other developers looking to use the the Embla library in their own product.

Best of luck to you!
